### PR TITLE
Backport #2116 to releases/2.5

### PR DESCRIPTION
**Pull request by mstanleyjones**
Improve formatting of configuration.md

Backported from master to release/2.5

Signed-off-by: Misty Stanley-Jones <misty@docker.com>

### CODE REVIEW

**Comment by dmcgowan** [comment]
Can you cherry pick your original commit rather than the merge commit?

**Comment by mstanleyjones** [comment]
I can, in `docker/docker` they prefer the merge commit. is there a way for me to redo it without having to resolve 39 conflicts again?

**Comment by dmcgowan** [comment]
@mstanleyjones no need to re-resolve, you can just edit the commit message. I have never seen it done that way in `docker/docker`. I prefer to have the original commit message as the cherry pick commit message and I find the merge commit message confusing given the context.

**Comment by mstanleyjones** [comment]
OK I changed the commit message.

File: docs/configuration.md [hunk 1]
```diff
@@ -1,457 +1,453 @@
...
+          password: password
+          insecure: true
+        from: sender@example.com
+        to:
+          - errors@example.com
+loglevel: debug # deprecated: use "log"
+storage:
+  filesystem:
+    rootdirectory: /var/lib/registry
+maxthreads: 100
```
Comment (dmcgowan) [inline docs/configuration.md]: This got moved, should be same level as rootdirectory

File: docs/configuration.md [hunk 2]
```diff
@@ -703,71 +613,25 @@ it supports any interesting structures desired, leaving it up to the middleware
...
+| `privatekey` | yes   | The private key for Cloudfront, provided by AWS.        |
+| `keypairid` | yes    | The key pair ID provided by AWS.                         |
+| `duration` | no      | An integer and unit for the duration of the Cloudfront session. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, or `h`. For example, `3000s` is valid, but `3000 s` is not. If you do not specify a `duration` or you specify an integer without a time unit, the duration defaults to `20m` (20 minutes).|
+
+### `redirect`
+
+You can use the `redirect` storage middleware to specify a custom URL to a
+location of a proxy for the layer stored by the S3 storage driver.
+
+|-----------|----------|-------------------------------------------------------------------------------------------------------------|
```
Comment (dmcgowan) [inline docs/configuration.md]: missing line for
```
| Parameter | Required | Description                                                                                                 |
```

File: docs/configuration.md [hunk 3]
```diff
@@ -703,71 +613,25 @@ it supports any interesting structures desired, leaving it up to the middleware
...
+| `baseurl` | yes      | `SCHEME://HOST` at which layers are served. Can also contain port. For example, `https://example.com:5443`. |
+
+## `reporting`
```
Comment (dmcgowan) [inline docs/configuration.md]: This must have been part of a different change in master, but in master this is surrounded by ` ``` `

File: docs/configuration.md [hunk 4]
```diff
@@ -1095,776 +758,277 @@ header's payload values.
...
+|-----------|----------|-------------------------------------------------------|
+| `addr`    | yes      | The TCP address and port to connect to.               |
+| `timeout` | no       | How long to wait before timing out the TCP connection. A positive integer and an optional suffix indicating the unit of time. The suffix is one of `ns`, `us`, `ms`, `s`, `m`, or `h`. If you specify a value but omit the suffix, the value is interpreted as a number of nanoseconds. |
+| `interval`| no       | How long to wait between repetitions of the check. A positive integer and an optional suffix indicating the unit of time. The suffix is one of `ns`, `us`, `ms`, `s`, `m`, or `h`. Defaults to `10s` if the value is omitted. If you specify a value but omit the suffix, the value is interpreted as a number of nanoseconds. |
+| `threshold`| no      | The number of times the check must fail before the state is marked as unhealthy. If this field is not specified, a single failure marks the state as unhealthy. |
+
+
+## `proxy`
 
     proxy:
```
Comment (dmcgowan) [inline docs/configuration.md]: Same with this section, is surrounded by ` ``` `

File: docs/configuration.md [hunk 5]
```diff
@@ -1095,776 +758,277 @@ header's payload values.
...
+middleware:
+  storage:
+  - name: cloudfront
+    disabled: false
+    options:
+      baseurl: http://d111111abcdef8.cloudfront.net
+      privatekey: /path/to/asecret.pem
+      keypairid: asecret
+      duration: 60
+
```
Comment (dmcgowan) [inline docs/configuration.md]: missing ` ``` `

**Comment by mstanleyjones** [comment]
Addressed the feedback. Will double-check in #2123 as well.

**Pull request merged**